Test cases are skipped on Windows - enabling them hits various failures (`mm_malloc.h` not found, runtime vector indices)

Currently, in `caseFromFile`: 

```zig
    var target: ?std.Target.Query = null;
    var skip_windows = true;
```

This seems unintended, as there is a path that sets this to true if `skip_windows=true` is present in the test case.

Setting this to `false` causes more tests to run, but they fail with various errors:


`x86_64-windows-gnu`: 
```
> zig build test -Dtarget=x86_64-windows-gnu
...

test
└─ test-run-translated
   └─ run-translated native typeof_operator
      └─ compile exe typeof_operator Debug native
         └─ translate-c native typeof_operator failure
C:\cygwin64\home\kcbanner\kit\zig\build-stage3-release\lib\zig\libc\include\any-windows-any\malloc.h:138:10: fatal error: 'mm_malloc.h' not found
#include <mm_malloc.h>
```


`x86_64-windows-msvc`
```
test
└─ test-run-translated
   └─ run-translated x86_64-windows-msvc vector_casting
      └─ compile exe vector_casting Debug x86_64-windows-msvc 1 errors
.zig-cache\o\fa3a1cc067248c39e855192c996a963d\x86_64-windows-msvc vector_casting.zig:373:45: error: vector index not comptime known
            if ((@as(c_int, unsigned_vector[@bitCast(@as(isize, @intCast(i)))]) != @as(c_int, @as(u8, @bitCast(@as(i8, signed_vector[@bitCast(@as(isize, @intCast(i)))]))))) and (@as(c_int, unsigned_vector[@bitCast(@as(isize, @intCast(i)))]) != @as(c_int, @as(u8, @bitCast(@as(i8, signed_vector[@bitCast(@as(isize, @intCast(@as(c_int, 7) - i)))])))))) {

...

test
└─ test-run-translated
   └─ run-translated x86_64-windows-msvc Wide,_UTF-16,_and_UTF-32_string_literals
      └─ compile exe Wide,_UTF-16,_and_UTF-32_string_literals Debug x86_64-windows-msvc 3 errors
error: lld-link: duplicate symbol: _Avx2WmemEnabledWeakValue
    note: defined at .zig-cache\o\0a0bc49d946d8a4f14ac783b54a05867\Wide,_UTF-16,_and_UTF-32_string_literals_zcu.obj
    note: defined at libucrt.lib(mbtowc.obj)
error: lld-link: duplicate symbol: _Avx2WmemEnabledWeakValue
    note: defined at .zig-cache\o\0a0bc49d946d8a4f14ac783b54a05867\Wide,_UTF-16,_and_UTF-32_string_literals_zcu.obj
    note: defined at libucrt.lib(mbrtowc.obj)
error: lld-link: duplicate symbol: _Avx2WmemEnabledWeakValue
    note: defined at .zig-cache\o\0a0bc49d946d8a4f14ac783b54a05867\Wide,_UTF-16,_and_UTF-32_string_literals_zcu.obj
    note: defined at libucrt.lib(write.obj)
error: the following command failed with 3 compilation error
```

--- a/build.zig.zon
+++ b/build.zig.zon
@@ -5,12 +5,12 @@
 
     .fingerprint = 0x1c84f6455a54f043, // Changing this has security and trust implications.
 
-    .minimum_zig_version = "0.16.0-dev.233+a0ec4e270",
+    .minimum_zig_version = "0.16.0-dev.577+c50aa2b95",
 
     .dependencies = .{
         .aro = .{
-            .url = "git+https://github.com/Vexu/arocc#90a03edeaa5ccb3b7d05da0be88b5905315b2ce2",
-            .hash = "aro-0.0.0-JSD1QsnDJwCjS7KTM0kLZYKPSdNKz-mNWKHDBVM3D4BO",
+            .url = "git+https://github.com/Vexu/arocc#6f7a5dd36545d35ffdbf7583a67596e3921769f8",
+            .hash = "aro-0.0.0-JSD1QmndJwCAV9Cyp5Qx6SsmSwjUCG_F3dqKgNRHpImr",
         },
     },
 

--- a/build/Translator.zig
+++ b/build/Translator.zig
@@ -108,6 +108,11 @@ pub fn initInner(
             for (libc.libc_framework_dir_list) |framework_dir| {
                 appendIncludeArg(run, "-iframework", .{ .cwd_relative = framework_dir });
             }
+
+            // Add Clang builtin includes with `-idirafter` to supplement missing headers
+            // like `mm_malloc.h` and `intrin.h` while still preferring Aro's own ones when they exist.
+            const clang_intrinsic_include_dir = b.graph.zig_lib_directory.join(b.graph.arena, &.{"include"}) catch @panic("OOM");
+            appendIncludeArg(run, "-idirafter", .{ .cwd_relative = clang_intrinsic_include_dir });
         }
     } else {
         run.addArg("-nostdinc");

--- a/src/Translator.zig
+++ b/src/Translator.zig
@@ -866,12 +866,18 @@ fn transVarDecl(t: *Translator, scope: *Scope, variable: Node.Variable) Error!vo
         break :blk null;
     };
 
+    // TODO actually set with @export/@extern
+    const linkage = variable.qt.linkage(t.comp);
+    if (linkage != .strong) {
+        try t.warn(scope, variable.name_tok, "TODO {s} linkage ignored", .{@tagName(linkage)});
+    }
+
     const alignment: ?c_uint = variable.qt.requestedAlignment(t.comp) orelse null;
     var node = try ZigTag.var_decl.create(t.arena, .{
         .is_pub = toplevel,
         .is_const = is_const,
         .is_extern = is_extern,
-        .is_export = toplevel and variable.storage_class == .auto,
+        .is_export = toplevel and variable.storage_class == .auto and linkage == .strong,
         .is_threadlocal = variable.thread_local,
         .linksection_string = linksection_string,
         .alignment = alignment,
@@ -1361,13 +1367,19 @@ fn transFnType(
         }
     };
 
+    // TODO actually set with @export/@extern
+    const linkage = func_qt.linkage(t.comp);
+    if (linkage != .strong) {
+        try t.warn(scope, source_loc, "TODO {s} linkage ignored", .{@tagName(linkage)});
+    }
+
     const payload = try t.arena.create(ast.Payload.Func);
     payload.* = .{
         .base = .{ .tag = .func },
         .data = .{
             .is_pub = ctx.is_pub,
             .is_extern = ctx.is_extern,
-            .is_export = ctx.is_export,
+            .is_export = ctx.is_export and linkage == .strong,
             .is_inline = ctx.is_always_inline,
             .is_var_args = switch (func_ty.kind) {
                 .normal => false,

--- a/test/cases.zig
+++ b/test/cases.zig
@@ -121,7 +121,7 @@ fn caseFromFile(b: *std.Build, entry: std.fs.Dir.Walker.Entry) !Case {
     };
 
     var target: ?std.Target.Query = null;
-    var skip_windows = true;
+    var skip_windows = false;
 
     var it = std.mem.tokenizeScalar(u8, manifest, '\n');
 

--- a/test/cases.zig
+++ b/test/cases.zig
@@ -49,6 +49,7 @@ pub fn lowerCases(
 
         const case_targets = if (case.target) |*t| t[0..1] else targets;
         for (case_targets) |case_target| {
+            if (case.skip_vector_index) continue;
             if (case.skip_windows and case_target.result.os.tag == .windows) {
                 continue;
             }
@@ -88,6 +89,7 @@ const Case = struct {
     input: []const u8,
     kind: Kind,
     skip_windows: bool,
+    skip_vector_index: bool,
 
     const Expect = enum { pass, fail };
 
@@ -122,6 +124,7 @@ fn caseFromFile(b: *std.Build, entry: std.fs.Dir.Walker.Entry) !Case {
 
     var target: ?std.Target.Query = null;
     var skip_windows = false;
+    var skip_vector_index = false;
 
     var it = std.mem.tokenizeScalar(u8, manifest, '\n');
 
@@ -146,6 +149,8 @@ fn caseFromFile(b: *std.Build, entry: std.fs.Dir.Walker.Entry) !Case {
             target = try .parse(.{ .arch_os_abi = value });
         } else if (std.mem.eql(u8, key, "skip_windows")) {
             skip_windows = std.mem.eql(u8, value, "true");
+        } else if (std.mem.eql(u8, key, "skip_vector_index")) {
+            skip_vector_index = std.mem.eql(u8, value, "true");
         } else return error.InvalidTestConfigOption;
     }
 
@@ -158,6 +163,7 @@ fn caseFromFile(b: *std.Build, entry: std.fs.Dir.Walker.Entry) !Case {
             .translate => .{ .translate = try trailingSplit(b.allocator, &it) },
         },
         .skip_windows = skip_windows,
+        .skip_vector_index = skip_vector_index,
     };
 }
 

--- a/test/cases/run/Ensure_while_loop_under_an_if_doesn't_steal_the_else.c
+++ b/test/cases/run/Ensure_while_loop_under_an_if_doesn't_steal_the_else.c
@@ -14,4 +14,3 @@ int main(void) {
 }
 
 // run
-// skip_windows=true

--- a/test/cases/run/Wide,_UTF-16,_and_UTF-32_character_literals.c
+++ b/test/cases/run/Wide,_UTF-16,_and_UTF-32_character_literals.c
@@ -15,4 +15,3 @@ int main() {
 }
 
 // run
-// skip_windows=true

--- a/test/cases/run/__builtin_convertvector.c
+++ b/test/cases/run/__builtin_convertvector.c
@@ -13,3 +13,4 @@ int main(int argc, char**argv) {
 }
 
 // run
+// skip_vector_index=true

--- a/test/cases/run/__builtin_shufflevector.c
+++ b/test/cases/run/__builtin_shufflevector.c
@@ -44,3 +44,4 @@ int main(int argc, char**argv) {
 }
 
 // run
+// skip_vector_index=true

--- a/test/cases/run/basic_vector_expressions.c
+++ b/test/cases/run/basic_vector_expressions.c
@@ -20,3 +20,4 @@ int main(int argc, char**argv) {
 }
 
 // run
+// skip_vector_index=true

--- a/test/cases/run/vector_casting.c
+++ b/test/cases/run/vector_casting.c
@@ -15,3 +15,4 @@ int main(int argc, char**argv) {
 }
 
 // run
+// skip_vector_index=true

